common/app/routes/challenges/utils.js

```diff
@@ -282,6 +282,16 @@ export function filterComingSoonBetaFromEntities(
   };
 }
 
+export function searchableChallengeTitles({ challenge: challengeMap } = {}) {
+  return Object.keys(challengeMap)
+    .map(dashedName => challengeMap[dashedName])
+    .reduce((accu, current) => ({
+        ...accu,
+        [current.dashedName]: current.title
+      })
+    , {});
```

Comment:
Using spread hear is quite slow and creates tons of objects needlessly.

Since the accumulator does not exist outside of the reduce function it is a ok to mutate it here and speed up this process.

```js
       .reduce((accu, current) => {
          accu[current.dashedName] = current.title;
          return accu;
       }, {})
```